File: Reversi/Factories/CellFactory.cs
using System;
using Reversi.Board;
using Reversi.Util;

namespace Reversi.Factories
{
    public class CellFactory
    {
        public Cell CreateCell(int x, int y)
        {
            return new Cell()
            {
                Position = new Coords(x, y)
            };
        }
    }
}

File: Reversi-Website/Static/src/js/modules/reversi/board/cell-factory.js
/**
 *
 * @constructor
 */
Reversi.CellFactory = function () {

    /**
     *
     * @param {Reversi.Coords} coords
     * @returns {Reversi.Cell}
     */
    this.createCell = (coords) => {
        let colour = (coords.x + coords.y) % 2 === 1 ? "red" : "brown"; //TODO make cell colours dynamic

        return new Reversi.Cell(coords, colour);
    };

};